org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/extensions/InjectionExtension.java

```diff
@@ -66,8 +74,23 @@ public void afterEach(ExtensionContext context) throws Exception {
 	}
 
 
+	private static InjectWith getInjectWith(Class<?> cls) {
+		InjectWith result = null;
+		while (cls != null) {
+			InjectWith iw = cls.getAnnotation(InjectWith.class);
+			if (iw != null) {
+				if( result != null ) {
+					throw new AssertionError("Mulitple annotations @InjectWith are found (outer/inner class). Only a single one is supported.");
```

Comment:
I'd go for `org.junit.jupiter.api.extension.ExtensionConfigurationException` instead.

Reply:
done